cellmlmanip/model.py

```diff
@@ -1041,6 +1043,53 @@ def _convert_variable_instance(self, original_variable, cf, units, direction, mo
 
         return new_variable
 
+    def remove_fixable_singularities(self, exclude=set()):
+        """
+        Removes removable singularities from the model equations and replaces these with a piecewise.
+
+        The process looks for equations of any of the following forms, where U is a function of V:
+        - `U / (exp(U) - 1.0)`
+        - `U / (1.0 - exp(U))`
+        - `(exp(U) - 1.0) / U`
+        - `(1.0 - exp(U)) / U`
+        It replaces these with a piecewise 1e-7 either side of U==0 drawing a stright line in the region.
+        For example `(V + 5)/(exp(V + 5) - 1)` becomes
+        `((fabs(-V - 5.0000000000000000) < fabs(-4.9999999000000000 / 2 - -5.0000001000000000 / 2))
+        ? -0.494049243462503*V - 1.4702462167574 : ((5.0 + V) / (-1.0 + exp(5.0 + V))))`
+
+        See for more details appendix B in:
+        [Johnstone, R. H. (2018). Uncertainty characterisation in action potential modelling for cardiac drug safety.
+        University of Oxford.](https://ora.ox.ac.uk/objects/uuid:0a28829c-828d-4641-bfb0-11193ef47195)
+
+        :param exclude: set of variables which will not be substituted in the evaluation.
+
+        This ensures their defining equations will remain.
+        """
+        from ._singularity_fixes import remove_fixable_singularities
+        assert isinstance(exclude, set), 'eclude is expected to be a set'
```

Comment:
```suggestion
        assert isinstance(exclude, set), 'exclude is expected to be a set'
```
Though this should probably be a `raise TypeError` as it's complaining about user input.